src/sentry/auth/helper.py

```diff
@@ -339,7 +340,15 @@ def _respond(
         return render_to_response(template, default_context, self.request, status=status)
 
     def _post_login_redirect(self) -> HttpResponseRedirect:
-        response = HttpResponseRedirect(auth.get_login_redirect(self.request))
+        url = auth.get_login_redirect(self.request)
+        if self.request.POST.get("op") == "newuser":
+            # add events that we can handle on the front end
+            provider = self.auth_provider.provider if self.auth_provider else None
+            params = {
+                "marketing_events": json.dumps({"event_name": "Sign Up", "event_label": provider})
```

Comment:
Would it be reasonable to name this `analytic_events` instead of marketing events, to try and avoid introducing "marketing" related things into the sentry project?

Reply:
@EvanPurkhiser the problem is that we don't want to send these events to other destinations since we are recording them on the backend already (Google Analytics is special). I don't think we can avoid having a concept of marketing in Sentry in general (see https://github.com/getsentry/sentry/pull/27851).